tests/test_simple.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+from typecheck import check, Tracker
+
+def test_check_func_single_arg():
+    @check(n=int)
+    def myf(n):
+        return n 
+
+    assert myf(30) == 30
+    with pytest.raises(Tracker) as e:
+        myf("string")
+
+
+def test_check_fun_many_args():
+    @check(n=int, s=str)
+    def myf(n, s):
+        return n, s 
+
+    assert myf(30, "string") == (30, "string")
+
+    with pytest.raises(Tracker) as e:
+        myf("string", 30)
+
+
+def test_check_instance_method():
+
+    class Human:
+        @check(self=object, age=int)
```

Comment:
Maybe self should be dropped from the checker decorator ? or replaced with 'Human' for example although this will be bad for inheritance. I would say drop the check on the self.